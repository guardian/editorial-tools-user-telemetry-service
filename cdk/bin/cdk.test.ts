import {OUTDIR_ENV} from "aws-cdk-lib/cx-api";

describe('The riff-raff output YAML', () => {
  it('matches the snapshot', () => {
    // outdir changes for every test execution and didn't want to change cdk.ts too much - so overriding in the test
    process.env[OUTDIR_ENV] = 'cdk.out';
    // eslint-disable-next-line -- this require is necessary because we must set env var before importing riffRaff
    expect(require('./cdk').riffRaff.toYAML()).toMatchSnapshot();
  });
});
