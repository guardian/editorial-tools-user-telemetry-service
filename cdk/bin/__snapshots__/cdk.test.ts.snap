// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The riff-raff output YAML matches the snapshot 1`] = `
"allowedStages:
  - PROD
  - CODE
deployments:
  lambda-upload-eu-west-1-flexible-user-telemetry-redrive-from-s3-lambda:
    type: aws-lambda
    stacks:
      - flexible
    regions:
      - eu-west-1
    app: user-telemetry-redrive-from-s3-lambda
    contentDirectory: user-telemetry-redrive-from-s3-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: redrive-from-S3-lambda.zip
    actions:
      - uploadLambda
  cfn-eu-west-1-flexible-telemetry-stack:
    type: cloud-formation
    regions:
      - eu-west-1
    stacks:
      - flexible
    app: telemetry-stack
    contentDirectory: cdk.out
    parameters:
      templateStagePaths:
        PROD: TelemetryStackPROD.template.json
        CODE: TelemetryStackCODE.template.json
    dependencies:
      - lambda-upload-eu-west-1-flexible-user-telemetry-redrive-from-s3-lambda
  lambda-update-eu-west-1-flexible-user-telemetry-redrive-from-s3-lambda:
    type: aws-lambda
    stacks:
      - flexible
    regions:
      - eu-west-1
    app: user-telemetry-redrive-from-s3-lambda
    contentDirectory: user-telemetry-redrive-from-s3-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: redrive-from-S3-lambda.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-flexible-telemetry-stack
  event-api-lambda:
    type: aws-lambda
    app: event-api-lambda
    contentDirectory: event-api-lambda
    parameters:
      prefixStack: false
      fileName: event-api-lambda.zip
      functionNames:
        - event-api-lambda-
    regions:
      - eu-west-1
    stacks:
      - flexible
  event-s3-lambda:
    type: aws-lambda
    app: event-s3-lambda
    contentDirectory: event-s3-lambda
    parameters:
      prefixStack: false
      fileName: event-s3-lambda.zip
      functionNames:
        - event-s3-lambda-
    regions:
      - eu-west-1
    stacks:
      - flexible
"
`;
